lib/features/mailbox_dashboard/presentation/mailbox_dashboard_view_web.dart

```diff
@@ -396,24 +393,170 @@ class MailboxDashBoardView extends GetWidget<MailboxDashBoardController> with Ne
   }
 
   Widget _buildSearchForm(BuildContext context) {
-    return (SearchAppBarWidget(context, _imagePaths, _responsiveUtils,
-            controller.searchQuery,
-            controller.searchFocus,
-            controller.searchInputController,
-            suggestionSearch: controller.suggestionSearch)
-        ..addDecoration(const BoxDecoration(
-            borderRadius: BorderRadius.all(Radius.circular(10)),
-            color: AppColor.colorBgSearchBar))
-        ..setMargin(const EdgeInsets.only(right: 10))
-        ..setHeightSearchBar(45)
-        ..setHintText(AppLocalizations.of(context).search_mail)
-        ..addOnCancelSearchPressed(() {
-          controller.disableSearch();
-          controller.dispatchAction(CancelSelectionAllEmailAction());
-        })
-        ..addOnClearTextSearchAction(() => controller.clearSearchText())
-        ..addOnTextChangeSearchAction((query) => controller.addSuggestionSearch(query))
-        ..addOnSearchTextAction((query) => controller.searchEmail(context, query)))
-      .build();
+    return Row(
+        children: [
+          buildIconWeb(
+              icon: SvgPicture.asset(
+                  imagePaths.icBack,
+                  color: AppColor.colorTextButton,
+                  fit: BoxFit.fill),
+              onTap: () {
+                controller.disableSearch();
+                controller.dispatchAction(CancelSelectionAllEmailAction());
+              }),
+          Expanded(child: Container(
+            decoration: const BoxDecoration(
+              borderRadius: BorderRadius.all(Radius.circular(10)),
+              color: AppColor.colorBgSearchBar),
+            height: 45,
+            child: TypeAheadFormFieldQuickSearch<PresentationEmail, RecentSearch>(
```

Comment:
How about move your modification in `flutter_typeahead` into `tmail-flutter`? IMO, it should be good for maintaining

Reply:
That is easy. But do you want to do that? I think we should put it in a separate library so everyone can use it.

Reply:
the search widget is too specific for our app only, so please move that to our code base.